src/nsidc/iceflow/itrf/converter.py

```diff
@@ -69,17 +71,50 @@ def transform_itrf(
             if not plate:
                 plate = plate_name(Point(chunk.longitude.mean(), chunk.latitude.mean()))
             plate_model_step = (
-                f"+step +inv +init={target_itrf}:{plate} +t_epoch={target_epoch} "
+                # Perform coordinate propagation to the given epoch using the
+                # provided plate motion model (PMM).  An example is given in the
+                # message of this commit:
+                # https://github.com/OSGeo/PROJ/commit/403f930355926aced5caba5bfbcc230ad152cf86
+                f"+step +init={target_itrf}:{plate} +t_epoch={target_epoch} "
             )
 
+        itrf_transformation_step = ""
+        if source_itrf != target_itrf:
+            # This performs a helmert transform (see
+            # https://proj.org/en/9.4/operations/transformations/helmert.html). `+init=ITRF2014:ITRF2008`
+            # looks up the ITRF2008 helmert transformation step in the ITRF2014
+            # data file (see e.g.,
+            # https://github.com/OSGeo/PROJ/blob/master/data/ITRF2014). The
+            # `+inv` reverses the transformation. So `+init=ITRF2014:ITRF2008`
+            # performs a helmert transform from ITRF2008 to ITRF2014.
+            itrf_transformation_step = f"+step +inv +init={target_itrf}:{source_itrf} "
```

Comment:
I want to think more about how we can support ITRF transformations more broadly. E.g., if someone passes in a `target_itrf` of `ITRF2008` and a `source_itrf` of `ITRF2014`, this will fail because there is no `ITRF2008` data file with a defined transformation to `ITRF2014`. But there is an `ITRF2014` data file with transformation parameters for `ITRF2008`. We'd just drop the `+inv` here to support that.

I think the reason we did it this way in `valkyrie` was probably because we always expected to go from older -> newer, and e.g., IS2 was planned to use ITRF2014.

Reply:
We could have some code that tests each way for validity. E.g., if `+init=ITRF2008:ITRF2014` fails, swap and add `+inv`: `+inv +init=ITRF2014:ITRF2008`.

This may not always work however. To make things a little more complicated, the data files that make the `+init=` approach work are limited to a selection of ITRFs and is dependent on the version of `proj` that gets installed.

The latest version of `proj` has a transformations file available for ITRF2020, for example, but the version I've been testing against does not.

A user could construct their own pipeline and pass in the helmert transformation parameters directly, or maybe we add that as an option here.

Reply:
WIP implementation: https://github.com/nsidc/iceflow/pull/46/